python/ray/parameter.py

```diff
@@ -94,7 +94,7 @@ def __init__(self,
                  redis_port=None,
                  redis_shard_ports=None,
                  object_manager_port=None,
-                 node_manager_port=None,
+                 node_manager_port=ray_constants.DEFAULT_NODE_MANAGER_PORT,
```

Comment:
Wouldn't this break pretty often if you restart your console or are trying to run two Python programs at once?

Reply:
If you restart your console, then ray should also get restarted, but yeah it would in the latter case. Then how about for drivers that start ray, we can first pick an unused port from the driver side?

Reply:
I find a lot of times the port doesn't get released for a while, which is the reason you sometimes need to pick a random port.

It might possibly work to pick an unused _random_ port on the driver side (there is a slight race condition that someone claims it before the raylet starts, but this is unlikely so probably acceptable).